docs/operations/ksm.md

```diff
@@ -0,0 +1,92 @@
+# KSM Management
+
+Kernel Samepage Merging ([KSM](http://www.linux-kvm.org/page/KSM))
+allows de-duplication of memory. KSM tries to find identical Memory Pages and merge
+those to free memory.
+>Further Information:  
+>- [KSM (Kernel Samepage Merging) feature](https://www.thomas-krenn.com/en/wiki/KSM_(Kernel_Samepage_Merging)_feature)  
+>- [Kernel Same-page Merging (KSM)](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_tuning_and_optimization_guide/chap-ksm)
+
+## Enabling KSM through KubeVirt CR
+KSM can be turned on in the nodes by `spec.configuration.ksmConfiguration` in the KubeVirt CR.  
+`ksmConfiguration` instructs in which nodes KSM will be enabled, exposing a `nodeLabelSelector`.  
+`nodeLabelSelector` is a [LabelSelector](https://github.com/kubernetes/apimachinery/blob/60180f072f73eafec72ef9f2c418a6bb1357d434/pkg/apis/meta/v1/types.go#L1195)
+and defines the filter, based on the node labels. If a node labels match the label selector term,
+then in that node, KSM will be enabled.  
+>**NOTE**  
+>If `nodeLabelSelector` is nil KSM will not be enabled in any nodes.  
+>Empty `nodeLabelSelector` will enable KSM for every node.  
+
+#### Examples:
+
+- Enabling KSM in nodes in which the hostname is `node01` or `node03`:
+```yaml
+spec:
+  configuration:
+    ksmConfiguration:
+      nodeLabelSelector:
+        matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - node01
+              - node03
+```
+
+- Enabling KSM in nodes with labels `kubevirt.io/first-label: true`, `kubevirt.io/second-label: true`:
+```yaml
+spec:
+  configuration:
+    ksmConfiguration:
+      nodeLabelSelector:
+        matchLabels:
+          kubevirt.io/first-label: "true"
+          kubevirt.io/second-label: "true"
+```
+
+- Enabling KSM in every node:
+```yaml
+spec:
+  configuration:
+    ksmConfiguration:
+      nodeLabelSelector: {}
+```
+
+
+## Annotation and restore mechanism
+ 
+In those nodes where KubeVirt enables the KSM via configuration, an annotation will be
+added (`kubevirt.io/ksm-handler-managed`).  
+This annotation is an internal record to keep track of which nodes are currently 
+managed by virt-handler, so that it is possible to distinguish which nodes should be restored
+in case of future ksmConfiguration changes.
+
+Let's imagine this scenario:
+1. there are 3 nodes in the cluster and one of them(`node01`) has KSM externally enabled;
+2. patch the KubeVirt CR adding a ksmConfiguration which enables ksm for `node02` and `node03`;
+3. after a while, patch again the KubeVirt CR deleting the ksmConfiguration;
```

Comment:
Formatting issue: you probably need a blank line before the list starts.

I did learn just yesterday that there is a VSCode extension for markdown lint and it is pretty comprehensive. It would catch this if you use VSCode.

Reply:
Done :)